chore: remove unused components/files

hofmann-felix opened the pull request:
In this PR, all components that are no longer needed are removed. Some old components only can be deleted later because they are currently still needed (e.g. components in the blog or the career form).

I decided to not change the folder structure in this PR (e.g. move all new components into the `new-components` folder) to avoid conflicts with other PRs.

hofmann-felix committed:
chore: remove unused components/files

ghost commented:
# [Gatsby Cloud](https://www.gatsbyjs.com) Build Report
**satellytes.com-main**

:tada: Your build was successful! [See the Deploy preview here.](https://satellytescommain21751-chorefolderstructure.gtsb.io)

## Build Details

[View the build logs here.](https://gatsbyjs.com/dashboard/f215130d-15e3-4176-9252-e13166fe48de/sites/674be368-adf0-4ac5-8fae-f9a4493adc51/builds/f5c59262-25ab-402a-8f96-10209886cdb0/details)

:clock1: Build time: **3m**

## Performance

### Lighthouse report

| Metric | Score |
| ----- | ---- |
| Performance |   :large_orange_diamond: 69   |
| Accessibility |   :green_heart: 92   |
| Best Practices |   :green_heart: 100   |
| SEO |   :large_orange_diamond: 71   |

:link: [View full report](https://build-f5c59262-25ab-402a-8f96-10209886cdb0.gtsb.io/reports/lighthouse/index.html)

georgiee reviewed (changes requested):
+ How did you decide to delete? Did you use some IDE feature or manually?
+ The button is something we wrote for V2. Why is it not used yet? (Is this something missing or is it really unused?)
+ Can we remove the entire icons folder by placing the icon component near the reamining users ? Is this only the blog or what's left?


I block this PR until clarified. Thanks for your effort 👍

hofmann-felix committed:
fix: add button

hofmann-felix commented:
> * How did you decide to delete? Did you use some IDE feature or manually?
> * The button is something we wrote for V2. Why is it not used yet? (Is this something missing or is it really unused?)
> * Can we remove the entire icons folder by placing the icon component near the reamining users ? Is this only the blog or what's left?

@georgiee Thanks for your comment 👍
- Yes I used the "Find usage" feature of Webstorm
- You are right, the button component is definitely still needed. I have added it again. 
Currently, other components mainly use the `LinkButton` from `links.tsx`. That still needs to be changed, I added it to our bucket list.
- I added the remaining icons to the appropriate components (mainly in `carrer-form`). A few are still left: 
  - `next-arrow.tsx` and `prev-arrow.tsx`: only used in pagination -> should be replaced with the new icon component
  - all social icons: they are used in navigation and social-panel. I am not sure how to deal with them
  - `arrow.tsx`: Is removed in https://github.com/satellytes/satellytes.com/pull/261
  - `chevron.tsx`: only used in language switch -> should be replaced with the new icon component
  - `swoosh.tsx`: used in the old typography and in the header: I am not sure how to deal with it too
  
I suggest a separate PR for `chevron.tsx`, `next-arrow.tsx` and `prev-arrow.tsx`

hofmann-felix committed:
chore: move remaining icons to parents folder

georgiee reviewed (approved):
Thanks for the clean up. Very much appreciated 🙌 If you can go ahead and handle the remaining icons that in a separate PR that would be nice

hofmann-felix committed:
Merge branch 'main' into chore-folder-structure
